### ISSUE TITLE
Auth global signOut not signing out the user locally

### ISSUE THREAD

**Issue by royjit** [aws-amplify/amplify-android#1472]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-android/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-android/discussions).

### Language and Async Model

Java

### Amplify Categories

Authentication

### Gradle script dependencies

<details>

```groovy
// Put output below this line
    implementation 'com.amplifyframework:aws-auth-cognito:1.24.0'

```

</details>

### Environment information

<details>

```
# Put output below this line
Welcome to Gradle 6.7.1!

Here are the highlights of this release:
 - File system watching is ready for production use
 - Declare the version of Java your build requires
 - Java 15 support

For more details see https://docs.gradle.org/6.7.1/release-notes.html


------------------------------------------------------------
Gradle 6.7.1
------------------------------------------------------------

Build time:   2020-11-16 17:09:24 UTC
Revision:     2972ff02f3210d2ceed2f1ea880f026acfbab5c0

Kotlin:       1.3.72
Groovy:       2.5.12
Ant:          Apache Ant(TM) version 1.10.8 compiled on May 10 2020
JVM:          15.0.2 (Amazon.com Inc. 15.0.2+7)
OS:           Mac OS X 11.5.2 x86_64

```

</details>

### Describe the bug

Global signout is throwing `NotAuthorizedException` if the token is invalid instead of signing out locally. Whereas in iOS the plugin sign out the user locally and return a success response.

Android has [code in place](https://github.com/aws-amplify/amplify-android/blob/602eef1fb7dd1e86082d62343d104d8375554069/aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AWSCognitoAuthPlugin.java#L1088) with similar logic as iOS. But the exception in Android is mapped to a different exception. ie instead of mapping to "com.amazonaws.services.cognitoidentityprovider.model.NotAuthorizedException" it is mapped to "com.amazonaws.services.cognitoidentity.model.NotAuthorizedException".

### Reproduction steps (if applicable)

1. Follow steps to configure auth in Android - https://docs.amplify.aws/lib/auth/signin/q/platform/android/
2. Sign in with same user into two devices
3. Signout Globally from one device. Signout should succeed.
4. Call `fetchAuthSession` in second device. session.isSigned = false.
4. Now signout Globally from the second device to see the NotAuthorizedException
5. Call `fetchAuthSession` in second device. session.isSigned = true.

### Code Snippet

```java
// Put your code below this line.
Amplify.Auth.signOut(
                AuthSignOutOptions.builder().globalSignOut(true).build(),
                () -> Log.i("AuthQuickstart", "Signed out globally"),
                error -> Log.e("AuthQuickstart", error.toString())
        );
```

### Log output

<details>

```
// Put your logs below this line


```

</details>

**Comment by Jordan-Nelson** [comment]
Proposed fix: Update the mapping of the exceptions to the correct NotAuthorizedException (from com.amazonaws.services.cognitoidentityprovider)

**Comment by AntoEko** [comment]
fixed here ? https://github.com/aws-amplify/amplify-android/pull/1473

**Comment by changxu0306** [comment]
This issue has been fixed by #1473 .
